controllers/base_controller.py

```diff
@@ -117,6 +117,17 @@ def _require_login(self, target_url="/"):
                 abort=True
             )
 
+    def _require_permission(self, permission):
+        self._require_login()
+        logging.info("logged in")
+        self._require_registration()
+        logging.info("registered")
+        if permission not in self.user_bundle.account.permissions:
+            return self.redirect(
```

Comment:
1. Redirect to homepage with a flash message stating they don't have permission
2. Redirect to previously-visited page (or homepage if no history) with flash message
3. Redirect to a page built specifically to tell the user that they're not authorized (like a 404 page)

There may be more, but those are a few.